Psychtoolbox/PsychBasic/DrawFormattedText2.m

```diff
@@ -333,10 +350,14 @@
 
 % string can contain HTML-like formatting commands. Parse them and turn
 % them into formatting indicators, then remove them from the string to draw
-[tstring,fmtCombs,fmts,switches,previous] = getFormatting(win,tstring,baseColor,resetStyle);
+[tstring,fmtCombs,fmts,switches,previous] = getFormatting(win,char(tstring),baseColor,resetStyle);

```

Comment:
Quick testing on Octave 4.2 showed that this char(tstring) cast doesn't work with Octave, as char() will map all unicodes > 255 to zero. The demo errors out.
What does work though is adding cstring=char(tstring) in getFormatting() to cast inside getFormatting(), then for the regexp() calls use regexp(cstring,...) instead of regexp(tstring,...). Only regexp() needs char() type, so tstring stays as double() and gets returned as double() from getFormatting() again.

Moving the char() cast this way makes the demo work with Unicode, showing three red japanese glyps, followed by the rest of the glyphs in black.

Downside - each char() cast prints some out-of-range warning. DrawFormattedText.m has the same problem. So we should avoid char() casts whenever possible. And maybe use proper warning off ; on pairs to suppress warnings at the right places.

Reply:
Ha, sorry, missed notification of your comment. I just tested on 4.4.1 and ran into the same issue, which i fixed in the essentially the same way :)